opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/validation/Xs2aValidator.java

```diff
@@ -56,6 +67,16 @@ public void validate(DelegateExecution exec, Object... dtosToValidate) {
         );
     }
 
+    private Set<FieldCode> getFieldsToIgnoreValidate(Xs2aContext context, Class invokerClass) {
+        Approach approach = context.aspspProfile().getPreferredApproach();
+        return context.getRequestScoped().getValidationRules().stream()
+                .filter(it -> it.getEndpointClassCanonicalName().equals(invokerClass.getCanonicalName()))
```

Comment:
Possible NPE here - you declared ' it.getEndpointClassCanonicalName()' as nullable (and it is nullable).
Also, the logic is:
```
null != it.getEndpointClassCanonicalName() &&  it.getEndpointClassCanonicalName().equals(invokerClass.getCanonicalName())
```
as we want to have rule that covers all classes if no exact class was specified

Reply:
I would say
`null == it.getEndpointClassCanonicalName() || it.getEndpointClassCanonicalName().equals(invokerClass.getCanonicalName())`